### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'best_model_path' when `checkpoint_callback` = False

### ISSUE THREAD

**Issue by lezwon** [PyTorchLightning/pytorch-lightning#2961]
## 🐛 Bug

Model does not complete training on TPU and error's out with error `AttributeError: 'NoneType' object has no attribute 'best_model_path'` when `checkpoint_callback=False`.

### To Reproduce

```python
trainer=Trainer(tpu_cores=8, max_epochs=5, checkpoint_callback=False, distributed_backend='tpu')
trainer.fit(model)  
```

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-6de8b6915beb> in <module>
      1 trainer = Trainer(tpu_cores=8, precision=16, max_epochs=5, checkpoint_callback=False, distributed_backend='tpu')
----> 2 trainer.fit(model)

/opt/conda/lib/python3.7/site-packages/pytorch_lightning/trainer/states.py in wrapped_fn(self, *args, **kwargs)
     32             if entering is not None:
     33                 self.state = entering
---> 34             result = fn(self, *args, **kwargs)
     35 
     36             # The INTERRUPTED state can be set inside the run function. To indicate that run was interrupted

/opt/conda/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py in fit(self, model, train_dataloader, val_dataloaders, datamodule)
   1058             self.accelerator_backend.setup()
   1059             self.accelerator_backend.train(model)
-> 1060             self.accelerator_backend.teardown(model)
   1061 
   1062         else:

/opt/conda/lib/python3.7/site-packages/pytorch_lightning/accelerators/tpu_backend.py in teardown(self, model)
     60 
     61         # transfer back the best path to the trainer
---> 62         self.trainer.checkpoint_callback.best_model_path = best_path
     63         # todo, pass also bets score
     64 

AttributeError: 'NoneType' object has no attribute 'best_model_path'
```

### Expected behavior

Should complete training.

### Additional context

Seems similar to https://github.com/PyTorchLightning/pytorch-lightning/issues/2547

**Comment by max0x7ba** [comment]
@lezwon When I pass `progress_bar_refresh_rate=0` to `Trainer`, `fit` and then call `predict` it still raises:
```
  File "/home/max/src/es/tune4.py", line 177, in test
    preds = trainer.predict(model, dataloaders=datamodule.val_dataloader())
  File "/home/max/anaconda3/envs/torch/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1025, in predict
    results = self.fit(model)
  File "/home/max/anaconda3/envs/torch/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 499, in fit
    self.dispatch()
  File "/home/max/anaconda3/envs/torch/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 543, in dispatch
    self.accelerator.start_predicting(self)
  File "/home/max/anaconda3/envs/torch/lib/python3.8/site-packages/pytorch_lightning/accelerators/accelerator.py", line 79, in start_predicting
    self.training_type_plugin.start_predicting(trainer)
  File "/home/max/anaconda3/envs/torch/lib/python3.8/site-packages/pytorch_lightning/plugins/training_type/training_type_plugin.py", line 122, in start_predicting
    self._results = trainer.run_predict()
  File "/home/max/anaconda3/envs/torch/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 817, in run_predict
    self.predict_loop.setup(model, max_batches, dataloaders)
  File "/home/max/anaconda3/envs/torch/lib/python3.8/site-packages/pytorch_lightning/trainer/predict_loop.py", line 58, in setup
    self.trainer._progress_bar_callback.on_predict_start(self.trainer, self.trainer.lightning_module)
AttributeError: 'NoneType' object has no attribute 'on_predict_start'
```

**Comment by rajdeep-biswas** [comment]
Hi, I would like to how was this resolved? @williamFalcon

**Comment by akihironitta** [comment]
Hi @rajdeep-biswas, this issue should be fixed in #2962. If you're still experiencing this issue, could you create a new GitHub issue with your environment info and a reproducible script?
